File: src/main/java/hilfling/backend/hilfling/model/Photo.java
package hilfling.backend.hilfling.model;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;

@Data
@Entity
@Table(name = "photo")
@Inheritance(strategy = InheritanceType.JOINED)
public class Photo extends BasePhoto implements Serializable {
    public Photo() {
    }

    public Photo(
            Boolean goodPicture,
            Motive motive,
            Place place,
            SecurityLevel securityLevel,
            Gang gang,
            PhotoGangBanger photoGangBanger
    ) {
        super(goodPicture, motive, place, securityLevel, gang, photoGangBanger);
    }

    @Column(name = "s_url") // Is nullable when analog photo
    private String smallUrl;

    @Column(name = "m_url") // Is nullable when analog photo
    private String mediumUrl;

    @Column(name = "l_url") // Is nullable when analog photo
    private String largeUrl;

}

File: src/main/kotlin/hilfling/backend/hilfling/model/Photo.kt
package hilfling.backend.hilfing.model

import me.liuwj.ktorm.database.Database
import me.liuwj.ktorm.entity.Entity
import me.liuwj.ktorm.entity.sequenceOf
import me.liuwj.ktorm.schema.*

interface Photo : BaseModel<Photo> {
    companion object : Entity.Factory<Photo>()

    var isGoodPicture: Boolean

    var smallUrl: String
    var mediumUrl: String
    var largeUrl: String

    // Foreign keys
    var motive: Motive
    var place: Place
    var securityLevel: SecurityLevel
    var gang: Gang
    var photoGangBanger: PhotoGangBanger
}

object Photos : BaseTable<Photo>("photo") {
    val isGoodPicture = boolean("is_good_picture").bindTo { it.isGoodPicture }

    val smallUrl = varchar("small_url").bindTo { it.smallUrl }
    val mediumUrl = varchar("medium_url").bindTo { it.mediumUrl }
    val largeUrl = varchar("large_url").bindTo { it.largeUrl }

    // Foreign keys
    val motive = int("motive_id").references(Motives){it.motive}
    val place = int("place_id").references(Places){it.place}
    val securityLevel = int("security_level_id").references(SecurityLevels){it.securityLevel}
    val gang = int("gang_id").references(Gangs){it.gang}
    val photoGangBanger = int("photo_gang_banger_id").references(PhotoGangBangers){it.photoGangBanger}
}

val Database.photos get() = this.sequenceOf(Photos)